numpy/core/src/multiarray/alloc.c

```diff
@@ -74,12 +71,19 @@ _npy_alloc_cache(npy_uintp nelem, npy_uintp esz, npy_uint msz,
 #ifdef _PyPyGC_AddMemoryPressure
         _PyPyPyGC_AddMemoryPressure(nelem * esz);
 #endif
-#ifdef HAVE_MADV_HUGEPAGE
+#ifdef NPY_OS_LINUX
         /* allow kernel allocating huge pages for large arrays */
         if (NPY_UNLIKELY(nelem * esz >= ((1u<<22u)))) {
             npy_uintp offset = 4096u - (npy_uintp)p % (4096u);
             npy_uintp length = nelem * esz - offset;
+#ifdef MADV_HUGEPAGE
             madvise((void*)((npy_uintp)p + offset), length, MADV_HUGEPAGE);
+#else
+            // Use code 14 (MADV_HUGEPAGE) if it isn't defined. This gives
+            // a chance of enabling huge pages even if build with linux
+            // kernel < 2.6.38
+            madvise((void*)((npy_uintp)p + offset), length, 14);
```

Comment:
Why not just define MADV_HUGEPAGES and let it go through a more similar code path

Reply:
Yes, thanks for the good idea, let me fix that.

Reply:
Done in the latest commits, let me know if that's ok with you.

Reply:
ok with me means nothing, just a suggestion ;)

Reply:
Well, if it's ok with you, it means that you probably don't disagree nor have an idea to make it even cleaner. :)